arm-notificationhubs/2016-03-01/swagger/notificationhubs.json

```diff
@@ -0,0 +1,1742 @@
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.NotificationHubs/namespaces/{namespaceName}/notificationHubs/{notificationHubName}/pnsCredentials": {
+            "post": {
+                "tags": [
+                    "NotificationHubs"
+                ],
+                "operationId": "NotificationHubs_GetPnsCredentials",
+                "description": "Lists the PNS Credentials associated with a notification hub .",
+                "parameters": [{
+                    "name": "resourceGroupName",
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "The name of the resource group."
+                }, {
+                    "name": "namespaceName",
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "The namespace name."
+                }, {
+                    "name": "notificationHubName",
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "The notification hub name."
+                }, {
+                    "$ref": "#/parameters/ApiVersionParameter"
+                }, {
+                    "$ref": "#/parameters/SubscriptionIdParameter"
+                }],
+                "responses": {
+                    "200": {
+                        "description": "",
+                        "schema": {
+                            "$ref": "#/definitions/NotificationHubResource"
+                        }
+                    }
+                }
+            }
+        }
+    },
+    "definitions": {
+        "CheckAvailabilityParameters": {
+            "properties": {
+                "name": {
+                    "type": "string",
+                    "description": "Gets or sets name"
+                },
+                "location": {
+                    "type": "string",
+                    "description": "Gets or sets location."
+                },
+                "tags": {
+                    "type": "object",
+                    "additionalProperties": {
+                        "type": "string"
+                    },
+                    "description": "Gets or sets tags."
+                },
+                "isAvailiable": {
+                    "type": "boolean",
+                    "description": "Gets or sets true if the name is available and can be used to create new Namespace/NotificationHub. Otherwise false."
+                }
+            },
+            "required": [
+                "name"
+            ],
+            "description": "Parameters supplied to the Check Name Availability for Namespace and NotificationHubs."
+        },
+        "CheckAvailabilityResource": {
+            "properties": {
+                "id": {
```

Comment:
the response for check availability can never be a resource. The user will be using this operation to check whether the name is available before creating a resource of that name. So how can this response have properties like id, type, location, tags. It is completely non-sensible.

Infact the response model should have a  property like reason that specifies "why something was not available" if it indeed wasn't.

Please take a look at how storage has modeled and designed a similar operation https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2016-01-01/swagger/storage.json#L434 and double check what your service is actually doing.